pkg/kv/kvserver/raft_transport.go

```diff
@@ -902,16 +912,7 @@ func (t *RaftTransport) startProcessNewQueue(
 			// DialNode already logs sufficiently, so just return.
 			return
 		}
-		batchCtx, cancel := context.WithCancel(ctx)
-		defer cancel()
-
-		stream, err := client.RaftMessageBatch(batchCtx) // closed via cancellation
-		if err != nil {
-			log.Warningf(ctx, "creating batch client for node %d failed: %+v", toNodeID, err)
-			return
-		}
-
-		if err := t.processQueue(q, stream, class); err != nil {
+		if err := t.processQueue(ctx, q, client, class); err != nil {
```

Comment:
Worth a comment explaining why we don't use streamCtx here. As is, one might be tempted to coalesce back to a single context.

Reply:
Done. The comment is inside `processQueue`.